src/js/providers/tracks-mixin.ts

```diff
@@ -985,4 +990,16 @@ function cuesMatch(cue1: TrackCue, cue2: TrackCue): boolean {
         JSON.stringify(cue1.value) === JSON.stringify(cue2.value);
 }
 
+function hideIfOldTrackSafari(tracks: TextTrackList, track: TextTrackLike, trackId: string): boolean {
+    if (Browser.safari) {
+        if (track.parsed && !tracks[trackId]) {
```

Comment:
Do we need the `!tracks[trackId]` check? I don't have a ton of context, but it looks like `tracks` is an array and `trackId` could be something like `subtitles4` so that value might usually be undefined.

Reply:
Yep. If we discover native tracks part way through a video we set text tracks again, so we want to make sure we keep the parsed ones if they're part of tracksById, as they belong to the current playlistitem